src/Nox.Generator/Infrastructure/Persistence/DbContext.template.cs

```diff
@@ -108,15 +113,70 @@ public override async Task<int> SaveChangesAsync(CancellationToken cancellationT
         try
         {
             HandleSystemFields();
+            await HandleDomainEvents();
             return await base.SaveChangesAsync(cancellationToken);
-
         }
         catch(DbUpdateConcurrencyException)
         {
             throw new Nox.Exceptions.ConcurrencyException($"Latest value of {nameof(IEntityConcurrent.Etag)} must be provided");
         }
     }
 
+    private async Task HandleDomainEvents()
+    {
+        var domainEvents =  GetPendingDomainEvents();
+        
+        await DispatchEvents(domainEvents);
+        
+        ClearDomainEvents();
+    }
+
+    private List<IDomainEvent> GetPendingDomainEvents()
```

Comment:
I think we are doing two different this in the same function and its complex to understand I Would Suggest something like this:

```
private async Task HandleDomainEvents()
    {
         var entitiesWithDomainEvents = GetEntitiesWithDomainEvents();

        RaiseDomainEventsFor(entitiesWithDomainEvents)

        await DispatchEvents(entitiesWithDomainEvents.SelectMany(entity => e.DomainEvents));

        entitiesWithDomainEvents.ForEach(entity => entity.ClearDomainEvents());
    }

public ICollection<IEntityHaveDomainEvents> GetEntitiesWithDomainEvents()
{
     return ChangeTracker.Entries<IEntityHaveDomainEvents>();
}
```

Reply:
Desired changes were applied